### PR TITLE
WIXFEAT:4773 Part 3 - Move OnPlan* into BAProc 

### PR DESCRIPTION
**Pull request by rseanhall**
https://github.com/wixtoolset/issues/issues/4773

### CODE REVIEW

File: src/burn/engine/plan.cpp [hunk 1]
```diff
@@ -1285,11 +1285,10 @@ extern "C" HRESULT PlanRelatedBundlesBegin(
 
         pRelatedBundle->package.defaultRequested = pRelatedBundle->package.requested;
 
-        int nResult = pUserExperience->pUserExperience->OnPlanRelatedBundle(pRelatedBundle->package.sczId, &pRelatedBundle->package.requested);
-        hr = UserExperienceInterpretResult(pUserExperience, MB_OKCANCEL, nResult);
-        ExitOnRootFailure(hr, "UX aborted plan related bundle.");
+        hr = UserExperienceOnPlanRelatedBundle(pUserExperience, pRelatedBundle->package.sczId, &pRelatedBundle->package.requested);
```
Comment (robmen) [inline src/burn/engine/plan.cpp:1301]: Curious if you're thinking about renaming `UserExperience` in these functions to BA too?

File: src/ext/BalExtension/wixstdba/wixstdba.mc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+; // .
+;
+; // MessageId=#
+; // Severity=Error
+; // SymbolicName=MSG_ERROR
+; // Language=English
+; // Error %1.
+; // .
+
+MessageId=1
```
Comment (robmen) [inline src/ext/BalExtension/wixstdba/wixstdba.mc:32]: These feel duplicated from Burn engine. I don't quite follow whey they need to be here.

File: src/burn/engine/plan.cpp (hunk 1 above)
Comment (rseanhall) [reply src/burn/engine/plan.cpp:1301]: I'm thinking about going through Burn and getting rid of UX, UserExperience, etc.  Not sure if I'll have the motivation, though :)

File: src/ext/BalExtension/wixstdba/wixstdba.mc (hunk 2 above)
Comment (rseanhall) [reply src/ext/BalExtension/wixstdba/wixstdba.mc:32]: They'll only show up if the BAFunctions was loaded.  It's supposed to be for the same reason that we do it in the engine - log what WixStdBA would have done so when someone complains about weird behavior we can point to the line in the log where their code changed something.

**Pull request closed without merging**